packages/react-graphql/src/tests/Query.test.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import gql from 'graphql-tag';
+import {createGraphQLFactory} from '@shopify/graphql-testing';
+
+import {Query} from '../Query';
+import {mountWithGraphQL} from '../hooks/tests/utilities';
+
+const petQuery = gql`
+  query PetQuery {
+    pets {
+      name
+    }
+  }
+`;
+
+const createGraphQL = createGraphQLFactory();
+const mockData = {
+  pets: [
+    {
+      __typename: 'Cat',
+      name: 'Garfield',
+    },
+  ],
+};
+
+describe('<Query />', () => {
+  it('returns the data', async () => {
+    const graphQL = createGraphQL({PetQuery: mockData});
+    const renderPropSpy = jest.fn(() => null);
+
+    await mountWithGraphQL(<Query query={petQuery}>{renderPropSpy}</Query>, {
```

Comment:
Nice! This is a great start. I'd also love to add a way to test the SSR issue that this component solves. But I don't think that is required for this PR.

Reply:
I could add a set of tests to ensure SSR behaviour. Do you feel these are implementation details of `useQuery` that are already tested though? https://github.com/Shopify/quilt/blob/master/packages/react-graphql/src/hooks/tests/query-ssr.test.tsx

Reply:
I was referring to the double call on ssr., but I think we have good coverage overall.